src/loggers/logger.service.ts

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Injectable, OnModuleInit } from "@nestjs/common";
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  HttpException,
+  HttpStatus,
+} from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import {
+  Logger,
+  LoggerConfig,
+  LoggerMethods,
+} from "./interfaces/logger.interface";
+import * as fs from "fs";
+
+@Injectable()
+export class ScicatLogger implements OnModuleInit {
+  private loggers: {
+    logger: Logger;
+    methods?: LoggerMethods;
+  }[] = [];
+  constructor(private configService: ConfigService) {}
+
+  async onModuleInit() {
+    let loggerConfigs = this.configService.get<LoggerConfig[]>("loggerConfigs");
+
+    const filePath = "loggers.json";
+    if (fs.existsSync(filePath)) {
+      const data = fs.readFileSync(filePath, "utf8");
+      loggerConfigs = JSON.parse(data);
+    }
+
+    if (!loggerConfigs || loggerConfigs.length < 1) {
+      console.log('No logger configs found in "loggers.json"');
+    } else {
+      await Promise.all(
+        loggerConfigs.map(async (loggerConfig) => {
+          try {
+            const LoggerClass = await import(loggerConfig.modulePath);
+            const logger = new LoggerClass.default(loggerConfig.config);
+
+            this.loggers.push({
+              logger: logger.getLogger(),
+              methods: loggerConfig.methods,
+            });
+          } catch (err) {
+            console.error(err);
+          }
+        }),
+      );
+    }
+  }
+
+  private handleLogForwarding(method: string, ...args: unknown[]): void {
```

Comment:
I was wondering why you are doing this dynamic mapping of the methods instead of declaring an abstract class and make sure that all the loggers implement such class?